File: ASCOM.Utilities/Alpaca Support/EntityClasses/AscomDevice.cs
using System;
using System.Collections.Generic;
using System.Net;
using System.Text;

namespace ASCOM.Utilities.Alpaca
{
    /// <summary>
    /// ASCOM device entity class
    /// </summary>
    public class AscomDevice
    {
        /// <summary>
        /// Default initialiser
        /// </summary>
        public AscomDevice() { }

        /// <summary>
        /// Initialise the ASCOM device name, ASCOM device type and ASCOM device unique ID, plus
        /// the Alpaca API device number, unique ID, device IP endpoint, Alpaca unique ID, interface version and status message
        /// </summary>
        /// <param name="ascomDdeviceName">ASCOM device name</param>
        /// <param name="ascomDeviceType">ASCOM device type</param>
        /// <param name="alpacaDeviceNumber">Alpaca API device number</param>
        /// <param name="ascomDeviceUniqueId">ASCOM device unique ID</param>
        /// <param name="ipEndPoint">ALpaca device IP endpoint</param>
        /// <param name="hostName">ALapca device host name</param>
        /// <param name="alpacaUniqueId">Alpaca device unique ID</param>
        /// <param name="interfaceVersion">Supported Alpaca interface version</param>
        /// <param name="statusMessage">ALapca device status message</param>
        public AscomDevice(string ascomDdeviceName, string ascomDeviceType, int alpacaDeviceNumber, string ascomDeviceUniqueId, IPEndPoint ipEndPoint, string hostName, string alpacaUniqueId, int interfaceVersion, string statusMessage)
        {
            AscomDeviceName = ascomDdeviceName;
            AscomDeviceType = ascomDeviceType;
            AlpacaDeviceNumber = alpacaDeviceNumber;
            AscomDeviceUniqueId = ascomDeviceUniqueId;
            IPEndPoint = ipEndPoint;
            HostName = hostName;
            AlpacaUniqueId = alpacaUniqueId;
            InterfaceVersion = interfaceVersion;
            StatusMessage = statusMessage;
        }

        /// <summary>
        /// ASCOM device name
        /// </summary>
        public string AscomDeviceName { get; set; }

        /// <summary>
        /// ASCOM device type
        /// </summary>
        public string AscomDeviceType { get; set; }

        /// <summary>
        /// Alpaca API device number
        /// </summary>
        public int AlpacaDeviceNumber { get; set; }

        /// <summary>
        /// ASCOM device unique ID
        /// </summary>
        public string AscomDeviceUniqueId { get; set; }

        /// <summary>
        /// Alpaca device IP endpoint
        /// </summary>
        public IPEndPoint IPEndPoint { get; set; }

        /// <summary>
        /// Alpaca device host name
        /// </summary>
        public string HostName { get; set; }

        /// <summary>
        /// Alapaca device unique ID
        /// </summary>
        public string AlpacaUniqueId { get; set; }

        /// <summary>
        /// SUpported Alpaca interface version
        /// </summary>
        public int InterfaceVersion { get; set; }

        /// <summary>
        /// Alpaca device status message
        /// </summary>
        public string StatusMessage { get; set; }
    }
}

File: ASCOM.Utilities/ASCOM.Utilities/AlpacaSupport/EntityClasses/AscomDevice.vb
Imports System.Net

Namespace ASCOM.Utilities.Support
    ''' <summary>
    ''' ASCOM device entity class
    ''' </summary>
    Public Class AscomDevice
        ''' <summary>
        ''' Default initialiser
        ''' </summary>
        Public Sub New()
        End Sub


        ''' <summary>
        ''' Initialise the ASCOM device name, ASCOM device type and ASCOM device unique ID, plus
        ''' the Alpaca API device number, unique ID, device IP endpoint, Alpaca unique ID, interface version and status message
        ''' </summary>
        ''' <paramname="ascomDdeviceName">ASCOM device name</param>
        ''' <paramname="ascomDeviceType">ASCOM device type</param>
        ''' <paramname="alpacaDeviceNumber">Alpaca API device number</param>
        ''' <paramname="uniqueId">ASCOM device unique ID</param>
        ''' <paramname="ipEndPoint">Alpaca device IP endpoint</param>
        ''' <paramname="hostName">ALapca device host name</param>
        ''' <paramname="alpacaUniqueId">Alpaca device unique ID</param>
        ''' <paramname="interfaceVersion">Supported Alpaca interface version</param>
        ''' <paramname="statusMessage">ALapca device status message</param>
        Public Sub New(ByVal ascomDdeviceName As String, ByVal ascomDeviceType As String, ByVal alpacaDeviceNumber As Integer, ByVal uniqueId As String, ByVal ipEndPoint As IPEndPoint, ByVal hostName As String, ByVal interfaceVersion As Integer, ByVal statusMessage As String)
            AscomDeviceName = ascomDdeviceName
            Me.AscomDeviceType = ascomDeviceType
            Me.AlpacaDeviceNumber = alpacaDeviceNumber
            Me.UniqueId = uniqueId
            Me.IPEndPoint = ipEndPoint
            Me.HostName = hostName
            Me.InterfaceVersion = interfaceVersion
            Me.StatusMessage = statusMessage
        End Sub


        ''' <summary>
        ''' ASCOM device name
        ''' </summary>
        Public Property AscomDeviceName As String

        ''' <summary>
        ''' ASCOM device type
        ''' </summary>
        Public Property AscomDeviceType As String

        ''' <summary>
        ''' Alpaca API device number
        ''' </summary>
        Public Property AlpacaDeviceNumber As Integer

        ''' <summary>
        ''' ASCOM device unique ID
        ''' </summary>
        Public Property UniqueId As String

        ''' <summary>
        ''' Alpaca device IP endpoint
        ''' </summary>
        Public Property IPEndPoint As IPEndPoint

        ''' <summary>
        ''' Alpaca device host name
        ''' </summary>
        Public Property HostName As String

        ''' <summary>
        ''' SUpported Alpaca interface version
        ''' </summary>
        Public Property InterfaceVersion As Integer

        ''' <summary>
        ''' Alpaca device status message
        ''' </summary>
        Public Property StatusMessage As String
    End Class
End Namespace
